### ISSUE TITLE
Feature: ALL, ANY, KEYS functions are required for some queries

### ISSUE THREAD

**Issue by Claus1** [kuzudb/kuzu#4092]
### API

Python

### Description

I attempted to perform a simple search for all node/link properties that should return any node with a property starting with a specific search string. I found that my query:
```cypher
MATCH (n)
WHERE ANY(key IN KEYS(n) WHERE String(n[key]) STARTS WITH '12')
RETURN n.*
```
is not supported. Could you please consider adding support for ANY, ALL, and KEYS to enable more concise queries?

**Comment by prrao87** [comment]
@Claus1, thanks for the report - seems like a usability issue for sure.

Just for reference, the query sample you posted above is a variant of this one that we do currently support via functions:

```cypher
MATCH (n)
WHERE size(list_filter(list, x->x STARTS WITH '12')) > 0
RETURN n.*
```
Could you give this query a try and check that the correct result is obtained? Granted, this version is not as readable, and harder to write, but we just wanted to check this this works on your end.

**Comment by Claus1** [comment]
@prrao87 For your query I got in console: Binder exception: Variable list is not in scope.

**Comment by prrao87** [comment]
> Binder exception: Variable list is not in scope.
`list` was just a placeholder. If you have just a few properties, you could specify a list like `['prop1', 'prop2', 'prop3']` and see if that works. Of course, the convenience feature of using `KEYS(n)` is something that will be added soon.

**Comment by andyfengHKU** [comment]
I believe @acquamarin is on this feature. We should have it at some point next week.

**Comment by Claus1** [comment]
@prrao87  Aa, sorry, for
```
MATCH (n: User) 
WHERE size(list_filter(['name'], x->x STARTS WITH '12')) > 0 
RETURN n.*
```
the error is:   Runtime exception: Trying to evaluate constant expression GREATER_THAN(SIZE(LIST_FILTER(LIST_CREATION(name),x->x STARTS WITH '12')),0) at runtime. This should be done at compile time.
```

**Comment by andyfengHKU** [comment]
Hi @Claus1,

Indeed it's a bug on our side, I have fixed it in #4098. Meanwhile I suggest waiting for @acquamarin to give a full support of the feature.

**Comment by andyfengHKU** [comment]
Hi @Claus1, this feature has been added in #4109. Now the following query should work directly.

```
MATCH (n) RETURN any(key IN keys(n) WHERE key STARTS WITH '12');
```

This feature is available on master or our nightly build version.
